src/ORM/DataObject.php

```diff
@@ -2014,10 +2014,8 @@ public function getComponents($componentName, $id = null)
         // Determine type and nature of foreign relation
         $details = $schema->getHasManyComponentDetails(static::class, $componentName);
         if ($details['polymorphic']) {
-            $result = PolymorphicHasManyList::create($componentClass, $details['joinField'], static::class);
-            if ($details['needsRelation']) {
-                Deprecation::withSuppressedNotice(fn () => $result->setForeignRelation($componentName));
-            }
+            $relation = $details['needsRelation'] ? $componentName : null;
+            $result = PolymorphicHasManyList::create($componentClass, $details['joinField'], static::class, $relation);
```

Comment:
`setForeignRelation()` is removed, and a new constructor arg is added in its place.